packages/machine-extractor/src/extractAction.ts

```diff
@@ -3,16 +3,17 @@ import { ActionNode } from './actions';
 
 // These types are copied over from studio blocks.
 // array and object are extracted as expressions so Studio render them correctly when exporting
-export type JsonEntry =
-  | { type: 'string'; value: string }
-  | { type: 'number'; value: number }
-  | { type: 'bigint'; value: string }
-  | { type: 'boolean'; value: boolean }
-  | { type: 'null'; value: null }
-  | { type: 'undefined'; value: undefined }
-  | { type: 'expression'; value: string };
-export type JsonValue = Pick<JsonEntry, 'value'>;
-type JsonType = Pick<JsonEntry, 'type'>;
+export type JsonItem =
+  | string
+  | number
+  | bigint
+  | boolean
+  | null
+  | undefined
```

Comment:
it's confusing to call this whole type (and related types) JSON since `undefined` is not a valid JSON value :P